drivers/adc/adc_stm32.c

```diff
@@ -533,17 +533,16 @@ static int adc_stm32_check_acq_time(uint16_t acq_time)
 }
 
 /*
- * Enable internal voltage reference source
+ * Enable internal channel source
  */
-static void adc_stm32_set_common_path(const struct device *dev)
+static void adc_stm32_set_common_path(const struct device *dev, uint32_t PathInternal)
 {
 	const struct adc_stm32_cfg *config =
 		(const struct adc_stm32_cfg *)dev->config;
 	ADC_TypeDef *adc = (ADC_TypeDef *)config->base;
 	(void) adc; /* Avoid 'unused variable' warning for some families */
 
-	LL_ADC_SetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc),
-				LL_ADC_PATH_INTERNAL_TEMPSENSOR);
+	LL_ADC_SetCommonPathInternalChAdd(__LL_ADC_COMMON_INSTANCE(adc), PathInternal);
```

Comment:
Good catch, but it seems there one more level of complexity involved here as `LL_ADC_SetCommonPathInternalChAdd` is not supported in some series and gets the CI failing.
Can you check that ?

Reply:
Thanks for the CI check, I thought it was again due to the NRF driver issue, so I didn't check further. I have just pushed a fix that should work on all series.